docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_testing-in-testing-farm.adoc

```diff
@@ -0,0 +1,51 @@
+= Adding an integration test
+
+link:https://docs.testing-farm.io/[Testing Farm] is a third-party test service that can be used as a test execution back-end for {ProductName}. This document guides you through an example of how to trigger Testing Farm tests from your {ProductName} *Integration tests*.
+
+.Prerequisites
+
+* You have created an application in {ProductName}.
+* You have a Testing Farm API key, which can be acquired by following the link:https://docs.testing-farm.io/Testing%20Farm/0.1/onboarding.html[Testing Farm onboarding guide].
+* You have have or are ready to create a git repository with link:https://fmf.readthedocs.io/[fmf] files suitable for use with the link:https://tmt.readthedocs.io/[tmt] testing tool. This is the testing technology used by Testing Farm under the hood. If you don't have a fmf tests already prepared, you can use our tmt link:https://github.com/ralphbean/tmt-hello-world[hello world repository].
+
+.Procedure
+
+You need to perform two major steps. Upload your Testing Farm API key, and register a *Integration test* to send requests to the Testing Farm API. Complete the following steps in the {ProductName} console:
+
+.Procedure - Upload your Testing Farm API key
+
+Follow similar instructions as in the xref:how-to-guides/configuring-builds/proc_creating-secrets-for-your-builds.adoc[Creating secrets for your builds] guide.
+
+. In {ProductName}, from the left navigation menu, select **Secrets**.
+. From the **Secrets** page, click **Add secret**.
+. From the **Add secret** page, choose **Build**.
+. Select secret type **Key/value secret**
+. For **Secret name**, enter `testing-farm-secret`.
+. Under **Key/value secret**, expand **Key/value 1**, then enter the key name `testing-farm-token`.
+NOTE: Be sure that the secret name is `testing-farm-secret` and the key name is `testing-farm-token`. The link:https://github.com/ralphbean/testing-farm-tekton/blob/main/tasks/testing-farm.yaml[tekton task] expects the secret and key name to be named these strings exactly.
+. For **Upload the file with value for your key or paste its contents**, paste the value of your token into the space under **Upload**.
+. Optional: Under **Labels**, add a label to tag or provide more context for your secret.
+. Click **Add secret**.
+
+.Procedure - Registering the Integration Test
+
+. Open an existing application and go to the *Integration tests* tab.
```

Comment:
https://github.com/redhat-appstudio/docs.appstudio.io/blob/main/docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_adding_an_integration_test.adoc

Maybe we may want to remove duplication in how to add integration test, and just refer to existing guide and only provide values for parameters?

Reply:
Yeah, I think for us - it's harder to maintain if this is duplicated, but for a user the guide is more useful if it tells you exactly what to do and you can follow it without having to click around.

Reply:
yup, if we manage to keep it consistent and valid :D 

I'm fine by keeping it as is then (cannot resolve comments here).